patches/PSP.cmake

```diff
@@ -1,34 +1,66 @@
-SET(CMAKE_SYSTEM_NAME Generic)
-SET(CMAKE_SYSTEM_VERSION 1)
-SET(CMAKE_CROSSCOMPILING TRUE)
+#
+# CMake toolchain file for PSP.
+#
+# Copyright 2019 - Wally
+# Copyright 2020 - Daniel 'dbeef' Zalega
 
-# set compiler
-set(CMAKE_C_COMPILER psp-gcc)
-set(CMAKE_CXX_COMPILER psp-g++)
+# Determine PSP toolchain installation directory (psp-config binary is added to path after successful installation):
+execute_process(COMMAND bash -c "psp-config --pspdev-path" OUTPUT_VARIABLE PSPDEV OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-SET(BUILD_SHARED_LIBS FALSE)
-SET(CMAKE_EXECUTABLE_SUFFIX ".elf")
+# Assert that PSP SDK path is now defined:
+if (NOT DEFINED PSPDEV)
+    message(FATAL_ERROR "PSPDEV not defined. Pass -DPSPDEV=/usr/local/pspdev or other installation location.")
+endif ()
 
-# set find root path
-execute_process(COMMAND psp-config --pspsdk-path
-  OUTPUT_VARIABLE PSPSDK_PATH
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
+# Set helper variables:
+set(PSPSDK ${PSPDEV}/psp/sdk)
+set(PSPBIN ${PSPDEV}/bin)
+set(PSPCMAKE ${PSPDEV}/psp/share/cmake)
 
-execute_process(COMMAND psp-config --psp-prefix
-  OUTPUT_VARIABLE PSP_PREFIX
-  OUTPUT_STRIP_TRAILING_WHITESPACE)
+# Basic CMake Declarations - Required (CMAKE_SYSTEM_NAME sets cross compiling to true)
+set(CMAKE_SYSTEM_NAME Generic)
+set(TOOLCHAIN "${PSPDEV}")
+set(CMAKE_C_COMPILER ${CC})
```

Comment:
Does this work? `CC` is set below.

Reply:
Turns out `CC` and the rest of toolchain variables (LD, AR, etc) was not needed after all.

Setting:

https://cmake.org/cmake/help/v3.9/variable/CMAKE_LANG_COMPILER.html#variable:CMAKE_%3CLANG%3E_COMPILER

For CXX and C is enough for CMake to find the rest of the tools. 
Checked this building my project with `-DCMAKE_VERBOSE_MAKEFILE` to see what is actually called.